### PR TITLE
Homework/andreyit

### PR DESCRIPTION
**Pull request by mailow7**

### CODE REVIEW

File: src/hoc/withAuthRedirect.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import {Redirect} from "react-router-dom";
+
+export const withAuthRedirect = (Component) => {
```
Comment (obabichev) [inline src/hoc/withAuthRedirect.js:4]: this class might be anonymous:
```
export const withAuthRedirect = (Component) => (class extends React.Component {
    render() {
        if (!this.props.isAuth) return <Redirect to="/login"/>
        return (<Component {...this.props} />)
    }
});
```

File: src/Components/Dashboard/DashboardContainer.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    name: state.auth.name
+
+});
+
+let mapDispatchToProps = (dispatch) => ({
+})
+
+let AuthRedirectComponent = withAuthRedirect(DashboardContainer)
+
+withRouter(AuthRedirectComponent);
```
Comment (obabichev) [inline src/Components/Dashboard/DashboardContainer.js:30]: looks like you dont's use it...

File: src/redux/auth-reducer.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+const authReducer = (state = initialState, action) => {
+
+    switch (action.type) {
+
+        case SET_USER_DATA:
+            return {
+                ...state,
+                ...action.payload
```
Comment (obabichev) [inline src/redux/auth-reducer.js:24]: You put here `...action.payload` but in the action actually 
```
{
    type: SET_USER_DATA,
    data: {accessToken, accessTokenExpiresIn, refreshToken, refreshTokenExpiresIn, isAuth}
}
```
data...payload...payload...data...

File: src/App.js [hunk 4]
```diff
@@ -1,50 +1,68 @@
...
-function App() {
-    const auth = localStorage.getItem('AUTH');
+    constructor(props) {
+        super(props);
 
-    return (
-        <div className='app-wrapper'>
 
+        this.state = {
+            isAuth: store.getState().auth.isAuth || null
```
Comment (obabichev) [inline src/App.js:27]: I suggest to use `mapStateToProps` and `connect` instead direct access to the store

Comment (obabichev) [reply src/App.js:27]: And this data is already in the redux store, you don't need to duplicate it inside compoentn

Comment (obabichev) [reply src/App.js:27]: Simple variant how to improve:

class App extends React.Component {
    render() {
        const {isAuth} = this.props;
        return (
            <Router>
                <div className='app-wrapper'>
                    <HeaderContainer/>

                    {!isAuth && <Switch>
                        <Route path='/login' component={LoginContainer}/>
                        <Route path="/register" component={RegisterContainer}/>
                        <Route path="/dashboard/:boardId?" component={DashboardContainer}/>
                        <Redirect to="/login"/>
                    </Switch>}

                    {!!isAuth && <Switch></Switch>}
                    <Footer/>
                </div>
            </Router>
        )

    }
}

const mapStateToProps = state => ({
    isAuth: !!state.auth.accessToken
});

export default compose(
    withRouter,
    connect(mapStateToProps, getAuthUserData()))(App);

File: src/hoc/withAuthRedirect.js (hunk 1 above)
Comment (mailow7) [reply src/hoc/withAuthRedirect.js:4]: fix

File: src/Components/Dashboard/DashboardContainer.js (hunk 2 above)
Comment (mailow7) [reply src/Components/Dashboard/DashboardContainer.js:30]: fix

File: src/redux/auth-reducer.js (hunk 3 above)
Comment (mailow7) [reply src/redux/auth-reducer.js:24]: fix

**Pull request merged**